scripts/component-templates.js

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable prettier/prettier */
+'use strict';
+
+const lowerFirst = require('lodash.lowerfirst');
+const snakeCase = require('lodash.snakecase');
+const toUpper = require('lodash.toupper');
+
+// component.tsx
+const component = (name) => `import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { TTheme } from '../../foundations/theme-provider/themes/theme.types';
+
+import ${name}Base from './${name}.base';
+import {
+    DEFAULT_${toUpper(snakeCase(name))}_SIZE,
+    ${toUpper(snakeCase(name))}_DEFINITIONS,
+} from './${name}.constants';
+import P${name} from './${name}.props';
+
+const ${name} = styled(${name}Base)<P${name}>(
+    ({
+        size = DEFAULT_${toUpper(snakeCase(name))}_SIZE,
+        theme,
+    }: ThemedStyledProps<P${name}, TTheme>): FlattenSimpleInterpolation => css\`
+        display: flex;
+        align-items: center;
+        justify-content: center;
+
+        color: $\{theme.text.primary};
+
+        width: $\{${toUpper(snakeCase(name))}_DEFINITIONS[size]}px;
+        height: $\{${toUpper(snakeCase(name))}_DEFINITIONS[size]}px;
+    \`
+);
+
+export default ${name};
+`;
+
+// component.base.tsx
+const base = (name) => `import React from 'react';
+
+import { DEFAULT_${toUpper(snakeCase(name))}_SIZE } from './${name}.constants';
+import P${name} from './${name}.props';
+
+const ${name}Base = ({ size = DEFAULT_${toUpper(snakeCase(name))}_SIZE }: P${name}): JSX.Element => (
+    <div>Hello 👋, I am a IconButton component with a default size of '{size}'.</div>
+);
+
+export default ${name}Base;
+`;
+
+// component.stories.mdx
+const story = (
+    name
+) => `import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+import { Utils, DEFAULT_ARGUMENTSTABLE_EXCLUSION } from '../../shared';
```

Comment:
nit: i know we keep changing that name but can we add the space between the ARGUMENTS_TABLE :D

Reply:
I will rename it in my next PR ... 👍🏼

Reply:
done! :D